src/mpi/romio/adio/ad_gpfs/ad_gpfs_aggrs.h

```diff
@@ -56,28 +56,17 @@ void ADIOI_GPFS_Calc_my_req(ADIO_File fd, ADIO_Offset * offset_list, ADIO_Offset
                             int **count_my_req_per_proc_ptr,
                             ADIOI_Access ** my_req_ptr, MPI_Aint ** buf_idx_ptr);
 
-    /*
-     * ADIOI_Calc_others_req
-     *
-     * param[in]  count_my_req_procs        Number of processes whose file domain my
-     *                                        request touches.
-     * param[in]  count_my_req_per_proc     count_my_req_per_proc[i] gives the no. of
-     *                                        contig. requests of this process in
-     *                                        process i's file domain.
-     * param[in]  my_req                    A structure defining my request
-     * param[in]  nprocs                    Number of nodes in the block
-     * param[in]  myrank                    Rank of this node
-     * param[out] count_others_req_proc_ptr Number of processes whose requests lie in
-     *                                        my process's file domain (including my
-     *                                        process itself)
-     * param[out] others_req_ptr            Array of other process' requests that lie
-     *                                        in my process's file domain
-     */
+void ADIOI_GPFS_Free_my_req(int nprocs, int *count_my_req_per_proc, ADIOI_Access * my_req);
+
 void ADIOI_GPFS_Calc_others_req(ADIO_File fd, int count_my_req_procs,
                                 int *count_my_req_per_proc,
                                 ADIOI_Access * my_req,
                                 int nprocs, int myrank,
-                                int *count_others_req_procs_ptr, ADIOI_Access ** others_req_ptr);
+                                int *count_others_req_procs_ptr,
+                                int **count_others_req_per_proc_ptr,
+                                ADIOI_Access ** others_req_ptr);
```

Comment:
@ashwinraghu Note that the signature of this function has been changed with the additional `count_others_req_per_proc_ptr` parameter.

Reply:
I see that, but looks like one of the callers in ad_gpfs_rdcoll.c  hasn't been changed.

Reply:
> 
> 
> I see that, but looks like one of the callers in ad_gpfs_rdcoll.c hasn't been changed.

I see now. I'll update it in a moment.

Reply:
@ashwinraghu Give it another try

Reply:
Breaks ad_lustre too:
../../../../src/mpi/romio/adio/ad_lustre/ad_lustre_wrcoll.c: In function 'ADIOI_LUSTRE_WriteStridedColl':
../../../../src/mpi/romio/adio/ad_lustre/ad_lustre_wrcoll.c:308:76: warning: passing argument 8 of 'ADIOI_Calc_others_req' from incompatible pointer type [-Wincompatible-pointer-types]
  308 |                           my_req, nprocs, myrank, &count_others_req_procs, &others_req);
      |                                                                            ^~~~~~~~~~~
      |                                                                            |
      |                                                                            ADIOI_Access ** {aka struct <anonymous> **}
In file included from ../../../../src/mpi/romio/adio/include/adio.h:509,
                 from ../../../../src/mpi/romio/adio/ad_lustre/ad_lustre.h:32,
                 from ../../../../src/mpi/romio/adio/ad_lustre/ad_lustre_wrcoll.c:14:
../../../../src/mpi/romio/adio/include/adioi.h:572:34: note: expected 'int **' but argument is of type 'ADIOI_Access **' {aka 'struct <anonymous> **'}
  572 |                            int **count_others_req_per_proc_ptr, ADIOI_Access ** others_req_ptr);
      |                            ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../../src/mpi/romio/adio/ad_lustre/ad_lustre_wrcoll.c:307:5: error: too few arguments to function 'ADIOI_Calc_others_req'
  307 |     ADIOI_Calc_others_req(fd, count_my_req_procs, count_my_req_per_proc,
      |     ^~~~~~~~~~~~~~~~~~~~~
In file included from ../../../../src/mpi/romio/adio/include/adio.h:509,
                 from ../../../../src/mpi/romio/adio/ad_lustre/ad_lustre.h:32,
                 from ../../../../src/mpi/romio/adio/ad_lustre/ad_lustre_wrcoll.c:14:
../../../../src/mpi/romio/adio/include/adioi.h:567:6: note: declared here
  567 | void ADIOI_Calc_others_req(ADIO_File fd, int count_my_req_procs,
      |      ^~~~~~~~~~~~~~~~~~~~~

Reply:
Let me fix it. Sorry for letting you be my warnings test, I don't have a system with the filesystem at the moment.

Reply:
@ashwinraghu Another try?

Reply:
That worked! Will get the bits onto a test system in a bit. Let you know how it goes soon.
Will also look at the changes meanwhile.

Reply:
Test results are looking good so far. I've run up to 1024 ranks, where a 64 rank job used hit the abort in #5339.
A 4096 ranks run timed out. Trying now with a longer timeout.

Reply:
A job of size ~2700 ranks also completed. Anything bigger is taking just too long, probably because of the huge file sizes and IO contention. This should do for now, I think.
Continuing to inspect the changes.